% Chapter 1

\chapter{Introduction} % Main chapter title

\label{Introduction} % For referencing the chapter elsewhere, use \ref{Chapter1} 

%----------------------------------------------------------------------------------------

% Define some commands to keep the formatting separated from the content 
\newcommand{\keyword}[1]{\textbf{#1}}
\newcommand{\tabhead}[1]{\textbf{#1}}
\newcommand{\code}[1]{\texttt{#1}}
\newcommand{\file}[1]{\texttt{\bfseries#1}}
\newcommand{\option}[1]{\texttt{\itshape#1}}

% \section{Introduction}
Deep learning methods have been widely used to tackle challenges in computational biology, including the recently succeed Alphafold2[3] which predict accurate protein structure given the protein sequence, alphafold achieved the accuracy by sophisticatedly network structure design and trained on large amount of training dataset, however, many questions in computational biology are presented in an unsupervised manner[4], where true labels are usually beyond the reach, thus supervised deep learning methods that require large amounts of labeling data are not directly applicable. Hybrid model that combine classical machine learning methods and deep learning are actively explored, one type of model applied classical clustering on a dimensional-reduced representation, the representation can be generated by a deep neural network through self-training including Denoise-Autoencoder[5] transformer-based encoder-decoder model[6], a process where a hidden representation is projected from the original or corrupted input sample, and the model is trained to reconstruct the input samples totally or partially from this projected hidden representation which is also called embedding. The model is trained by minimizing the reconstruction error so the underlying distribution of input samples are captured by the embedding of input samples, and the embedding can be used as the input with few-shot learning when there are very few labels or with clustering method when there is no label. Another way is through generative models such as probabilistic graphical models, probabilistic graph models are hard to apply as they suffer from intractable exact posterior inference, variational inference is used with a posterior distribution usually approximated by a neural network[7]. Hybrid model have achieved promising result in unsupervised and semi-supervised biological task, for example, Autoencoder-style models are used to generate low-dimensional gene embedding for single-cell RNA expression data[8-14], the gene embedding generated are then be used in afterward tasks such as cell type clustering[15]. In this thesis we designed and applied several new hybrid models to solve several different biological tasks including cell clustering on spatial transcriptomics data, post transcriptional modification detection on long read sequencing platform and drug discovery. 
\section{Cell clustering}
Cell clustering is a process where the cell is assigned to several groups based on their gene expression profile, itâ€™s a fundamental biological task and is requested by many downstream analyses in single-cell RNA sequencing[16]. Recent advances in Fluorescence in-situ Hybridization (FISH) technique enable recording a single cell level spatial transcriptomics for large numbers of cells[1,2,17], however, scRNA pipeline is usually used to analyze this data where spatial information is not taken into account when conduct cell clustering, so we developed a mixture model with denoise autoencoder embedding called FICT which combines both expression and neighborhood information when assigning cell type.
\section{RNA modification detection using Oxford Nanopore sequencing}
RNA modification played an important role in various biological processes including stem cell differentiation and renewal, brain function, immunity and cancer progression [18], among the several RNA modification, N6-Methyladenosine (m6A) is one of the most abundant modification, involved in mRNA expression, splicing, nuclear export, translation efficiency, RNA stability and miRNA processing[18]. Among several m6A detection methods, long read sequencing using Oxford Nanopore sequencer is a way that can give qualitative information about the whole m6A methylome, and with the potential to detect single-molecule read level modification, however it requires optimized detection methods to call the m6A information from the subtle signal change. There are several m6A datasets available, In-vitro transcription dataset is made from synthetic sequence [19,20], by introducing only canonical adenine or modified adenine when doing in-vitro transcription, all modified or non-modified datasets are created. However, training a basecaller directly on the non-or-whole modified read would fail on basecall modified state on reads sequencing from real biological samples, as the modification state is usually mixed in one read. So several methods address this problem by training a local kmer model, where a classifier is trained to call modification on segmented signal segments. The performance of the trained classifier relies on a good post-segmentation of the sequencing signal which is usually done by a HMM [21], whose performance is limited by its parameters learning from canonical RNA sequencing. Data generated from antibody capturing techniques [22,23] can only provide site-level modifications whereas read-level modifications are unknown. We developed a new kind of hybrid model where a NHMM is trained semi-supervised by conditioning on the output from a deep learning network, the modified and canonical reads are then segmented by the trained NHMM and added into a graph, reads with mixing modification state then sampled from the graph to produce a training dataset, this step can be seen as a data augmentation that eliminate the inductive bias when trained using homogeneous modified read, we then trained a new deep learning model on this dataset which give accurate methylation basecalling.
\section{Drug discovery with deep learning}
Drug discovery is one of the most important 
Related Biological technology
